client/src/containers/App/routes/private.routes.js

```diff
@@ -16,7 +20,19 @@ const PrivateRoutes = props => (
           {props.userState.token ? (
             <EditUserProfile {...routesProps} />
           ) : (
-            <LoginRedirection />
+            <LoginRedirection {...routesProps} />
+          )}
+        </div>
+      )}
+    />
+    <Route
+      path="/events/:event_id"
+      render={routeProps => (
+        <div>
+          {props.userState.token ? (
```

Comment:
@Sarmad93 we can access the event detail page without logging in. But when we click the button we need to redirect to the login page if we are not logged in